packages/amplify-graphql-model-transformer/publish-notification-lambda/src/index.ts

```diff
@@ -0,0 +1,100 @@
+import { LambdaClient, ListLayerVersionsCommand, ListLayerVersionsCommandOutput } from "@aws-sdk/client-lambda";
+import { PublishCommand, SNSClient } from "@aws-sdk/client-sns";
+
+const MAX_ITEMS = 50;
+const { LAYER_NAME, SNS_TOPIC_ARN, SNS_TOPIC_REGION } = process.env;
+
+const SUPPORTED_REGIONS = [
+  'us-east-1',
+  'us-west-2',
+  'us-west-1',
+  'eu-north-1',
+  'ap-south-1',
+  'eu-central-1',
+  'eu-west-1',
+  'sa-east-1',
+  'ap-southeast-1',
+  'ap-northeast-2',
+  'eu-west-2',
+  'us-east-2',
+  'ap-northeast-1',
+  'eu-west-3',
+  'ap-southeast-2',
+  'ca-central-1',
+  'ap-northeast-3',
+];
+
+type LayerConfig = {
+  layerArn: string | undefined;
+  region: string;
+};
+
+const getLatestVersion = async (layerName: string, region: string): Promise<LayerConfig> => {
+  const client = new LambdaClient({ region });
+  const versions = [];
+  let response: ListLayerVersionsCommandOutput | undefined;
+  do {
+    const command = new ListLayerVersionsCommand({
+      LayerName: layerName,
+      MaxItems: MAX_ITEMS,
+      Marker: response?.NextMarker,
+    });
```

Comment:
Is there any way we can perform a sort here so we don't need to page? (Or can page until we find a first element or something)? I'm worried about this growing over time, and theoretically timing out the querying lambda (unlikely, but we should avoid continually expanding the runtime here if possible)

Reply:
It returned the latest version as first element every time in my testing. But some blogs warned that this sorting behavior isn't consistent. This lambda lives in our service account so that it can be tweaked later if we run into any issues without impacting customers.